docs: update CONTRIBUTING.md regarding deprecation of `develop` (#4280)
algolia/instantsearch.js: CONTRIBUTING.md

You will find a playground for the widgets: https://**netlify-deploy-url**/stories.

> - Example: https://deploy-preview-3376--instantsearchjs.netlify.com/stories/

Reviewer: If there's now only one item in the list, we can inline it in the line above.

Author: https://github.com/algolia/instantsearch.js/pull/4280/commits/251433a6a8e6e6a71a1585a3f1c9e8fa4625e449

### Main version

> For the main version, go on `master` (`git checkout master`) and use:

Reviewer: ```suggestion
To release a stable version, go on `master` (`git checkout master`) and use:
```

### Main version

> For the main version, go on master (`git checkout master`) and use:

Reviewer: ```suggestion
For the main version, go on `master` (`git checkout master`) and use:
```

npm run release
```

> You can also release a beta version from another branch, using the beta flag in the command line:

Reviewer: ```suggestion
You can release beta versions from any branches, using the beta flag in the command line:
```

Reviewer 2: ```suggestion
You can release beta versions from any branch, using the beta flag in the command line:
```

npm run release
```

> You can also release a beta version from `next`, using the beta flag in the command line:

Author: `next` instead of `develop` makes sense here?

Reviewer: `next` makes sense for the next major version, we can add a mention in "Branch organization".

However, I think this sentence doesn't make sense anymore. We should be able to release a beta version from any branch with Ship.js.

Reviewer 2: ```suggestion
You can also release a beta version from another branch, using the beta flag in the command line:
```

?
